src/stdlib_io_disp.fypp

```diff
@@ -0,0 +1,181 @@
+            !! Brief Print.
+            write(unit_, fmt_c) (format_to_string(value(i), format_), i=1, 3), &
+                          '...', format_to_string(value(m), format_)
+        else
+            !! Full Print.
+            write(unit_, fmt_c) (format_to_string(value(i), format_), i=1, m)
+        end if
+    end procedure disp_1_${type[0]}$${kind}$
+
+    module procedure disp_2_${type[0]}$${kind}$
+        !! Display a ${type}$ matrix.
+        integer :: unit_
+        logical :: brief_
+        character(1) :: colon(5)
+        integer :: i, j, m, n
+        unit_  = merge(unit, output_unit, present(unit))
+        brief_ = merge(brief, .false., present(brief))
+        m = size(value, 1)
+        n = size(value, 2)
+
+        if(present(header)) write(unit_, *) header
+        write(unit_, *) '[matrix size: '//to_string(m)//'×'//to_string(n)//']'
+        if(brief_ .and. (m > 5 .or. n > 5)) then
+            !! Brief Print.
+            if(m > 5 .and. n > 5) then
+                do i = 1, 3
+                    write(unit_, fmt_c) (format_to_string(value(i,j), format_), j=1, 3), &
+                                  '...', format_to_string(value(i,n), format_)
+                end do
+                write(unit_, fmt_c) ':', ':', ':', ':', ':'
+                write(unit_, fmt_c) (format_to_string(value(m,j), format_), j=1, 3), &
+                              '...', format_to_string(value(m,n), format_)
+            elseif(m > 5 .and. n <= 5) then
+                do i = 1, 3
+                    write(unit_, fmt_c) (format_to_string(value(i,j), format_), j=1, n)
+                end do
+                colon(1:n) = ':'
+                write(unit_, fmt_c) colon(1:n)
+                write(unit_, fmt_c) (format_to_string(value(m,j), format_), j=1, n)
+            elseif(m <= 5 .and. n > 5) then
+                do i = 1, m
+                    write(unit_, fmt_c) (format_to_string(value(m,j), format_), j=1, 3), &
+                                  '...', format_to_string(value(m,n), format_)
+                end do
+            end if
+        else
+            !! Full Print.
+            do i = 1, m
+                write(unit_, fmt_c) (format_to_string(value(i,j), format_), j=1, n)
+            end do
+        end if
+    end procedure disp_2_${type[0]}$${kind}$
+    #:endfor
+
+    module procedure disp_character
+        !! Display a `character` scalar.
+        character(:), allocatable :: value_
+        integer :: unit_
+        value_ = optval(value, '')
+        unit_  = merge(unit, output_unit, present(unit))
+        if(present(header)) write(unit_, *) header
+        write(unit_, *) value_
+    end procedure disp_character
+
+    module procedure disp_string_type
+        !! Display a `string_type` scalar
+        integer :: unit_
+        unit_ = merge(unit, output_unit, present(unit))
+        if(present(header)) write(unit_, *) header
+        write(unit_, *) char(value)
+            !!\TODO: Need to improve ?
```

Comment:
The companion function `char` of `string_type` type is used here, which is convenient for the use of `write(unit, *)` in multiple scenarios, and may reduce the speed a little.